redirect.module

```diff
@@ -240,11 +240,11 @@ function redirect_delete_by_path($path, $langcode = NULL, $match_subpaths_and_re
   $query = db_select('redirect');
   $query->addField('redirect', 'rid');
   $query_or = db_or();
-  $query_or->condition('redirect_source__url', db_like($path), 'LIKE');
+  $query_or->condition('redirect_source__uri', db_like($path), 'LIKE');
   if ($match_subpaths_and_redirect) {
-    $query_or->condition('redirect_source__url', db_like($path . '/') . '%', 'LIKE');
-    $query_or->condition('redirect_redirect__url', db_like($path), 'LIKE');
-    $query_or->condition('redirect_redirect__url', db_like($path . '/') . '%', 'LIKE');
```

Comment:
Oh, just ignore me

On Sun, Feb 1, 2015 at 6:08 PM, Sascha Grossenbacher <
notifications@github.com> wrote:

> In redirect.module
> <https://github.com/md-systems/redirect/pull/6#discussion_r23902563>:
>
> >    if ($match_subpaths_and_redirect) {
> > -    $query_or->condition('redirect_source__url', db_like($path . '/') . '%', 'LIKE');
> > -    $query_or->condition('redirect_redirect__url', db_like($path), 'LIKE');
> > -    $query_or->condition('redirect_redirect__url', db_like($path . '/') . '%', 'LIKE');
>
> It is, but STARTS_WITH is an entity query only concept, because this is
> sql specific. This can be converted into an entity query eventually when we
> have support for base fields with multiple columns, but there are cases
> where it is not so easily possible, like the one where we join the watchdog
> table to search for 404's. The best we can do there is move it to a method
> in the storage class I guess.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/md-systems/redirect/pull/6/files#r23902563>.
>